src/sqlparser/tests/parser_test.rs

```diff
@@ -0,0 +1,199 @@
+                return Err(anyhow!(
+                    "Expected formatted ast:\n  {}\nActual:\n  {}",
+                    style(&expected_formatted_ast).green(),
+                    style(&formatted_ast).red(),
+                ));
+            }
+        }
+    }
+    Ok(())
+}
+
+fn run_test_file(file_name: &str) -> Result<()> {
+    let file_content = std::fs::read_to_string(file_name).unwrap();
+    let cases: Vec<TestCase> =
+        serde_yaml::from_str(&file_content).context("failed to parse yaml")?;
+    let mut error_msg = String::new();
+    for c in cases {
+        let input = c.input.clone();
+        if let Err(e) = run_test_case(c) {
+            write!(&mut error_msg, "With input:\n  {}\n{}\n", input, e)?;
+        }
+    }
+    if !error_msg.is_empty() {
+        bail!("{}", error_msg);
+    }
+    Ok(())
+}
+
+fn update_test_file(file_name: &str) -> Result<()> {
+    let file_content = std::fs::read_to_string(file_name)?;
+
+    let cases: Vec<TestCase> = serde_yaml::from_str(&file_content)?;
+
+    let mut new_cases = Vec::with_capacity(cases.len());
+
+    for case in cases {
+        let input = &case.input;
+        let ast = Parser::parse_sql(input);
+        let actual_case = match ast {
+            Ok(ast) => {
+                let [ast]: [Statement; 1] = ast
+                    .try_into()
+                    .expect("Only one statement is supported now.");
+
+                let actual_formatted_sql = case.formatted_sql.as_ref().map(|_| format!("{}", ast));
+                let actual_formatted_ast =
+                    case.formatted_ast.as_ref().map(|_| format!("{:?}", ast));
+
+                TestCase {
+                    input: input.clone(),
+                    formatted_sql: actual_formatted_sql,
+                    formatted_ast: actual_formatted_ast,
+                    error_msg: None,
+                }
+            }
+            Err(err) => {
+                let actual_error_msg = format!("{}", err);
+                TestCase {
+                    input: input.clone(),
+                    formatted_sql: None,
+                    formatted_ast: None,
+                    error_msg: Some(actual_error_msg),
+                }
+            }
+        };
+
+        if actual_case != case {
+            println!("{}\n{}\n", style(&case).red(), style(&actual_case).green())
+        }
+
+        new_cases.push(actual_case);
+    }
+
+    let mut output_file = File::create(file_name)?;
+    writeln!(
+        output_file,
+        "# This file is automatically generated by `src/sqlparser/tests/parser_test.rs`."
+    )?;
+    write!(output_file, "{}", serde_yaml::to_string(&new_cases)?)?;
+    Ok(())
+}
+
+/// The entry point of `test_runner`.
+fn main() {
+    let run_tests_args = &Arguments::from_args();
+    let mut tests = vec![];
+    let update = std::env::var("UPDATE_PARSER_TEST").is_ok();
+    std::env::set_current_dir("tests/testdata").unwrap();
+
```

Comment:
What about relying on `expect_test` to update result?
https://github.com/risingwavelabs/risingwave/blob/e16812bf3cbab2cdb59e47e62852197655996129/src/frontend/planner_test/src/lib.rs#L95-L98

Reply:
I find that it's not a good idea to use `expect_test` as it can not show diff one case at a time. Multiple cases would mix together. I'm going to revert this.

Reply:
It's a known issue https://github.com/risingwavelabs/risingwave/issues/10940 🤪

Reply:
I mainly just blindly update the tests and rely on git diff to see what's changed.